server/src/main/java/org/elasticsearch/cluster/block/ClusterBlockException.java

```diff
@@ -110,12 +111,18 @@ private static String buildMessageForIndexBlocks(Map<String, Set<ClusterBlock>>
     @Override
     public RestStatus status() {
         RestStatus status = null;
+        boolean onlyRetryableBlocks = false;
```

Comment:
I think this does not work as expected. Maybe you can add a test that has multiple blocks in it (using the `ElasticsearchAssertions.assertBlocked` maybe)?

As far as I can see, if you have two blocks in the exception, one with 403 and one with 429, this method still returns 429?

Reply:
OK, I will resolve it and add a tesst.

Reply:
I have fixed the bug. The change are:

1. If we have only 429, then we return 429.
2.If we have not only 429, but also 403 or other status code, then we return the maximum one.
I have added some tests for this change.